tests/integration/test_storage_dict/test.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+
+from helpers.cluster import ClickHouseCluster
+
+uuids = []
+
+
+@pytest.fixture(scope="module")
+def cluster():
+    try:
+        cluster = ClickHouseCluster(__file__)
+        cluster.add_instance(
+            "node1", main_configs=["configs/conf.xml"], with_nginx=True
+        )
+        cluster.start()
+
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+def test_storage_dict(cluster):
+    node1 = cluster.instances["node1"]
+
+    node1.query(f"insert into table function url(urldb) values ('foo', 'bar')")
+    result = node1.query(f"select * from url(urldb)")
+    assert result.strip() == "foo\tbar"
+
+    node1.query(
+        f"create dictionary dict (k String, v String) primary key k source(http(name urldict)) layout(complex_key_hashed()) lifetime(min 0 max 100)"
```

Comment:
Let's also add a test when dict arguments override config values like `source(http(name urldict format csv))`.
It should work.